packages/node/src/integrations/tracing/nest/sentry-nest-instrumentation.ts

```diff
@@ -163,33 +163,84 @@ export class SentryNestInstrumentation extends InstrumentationBase {
 
             target.prototype.intercept = new Proxy(target.prototype.intercept, {
               apply: (originalIntercept, thisArgIntercept, argsIntercept) => {
-                const [executionContext, next, args] = argsIntercept;
-                const prevSpan = getActiveSpan();
+                const context: MinimalNestJsExecutionContext = argsIntercept[0];
+                const next: CallHandler = argsIntercept[1];
+                const request = context.switchToHttp().getRequest();
```

Comment:
h: What happens if you are in an rpc context and we call this?

Reply:
My understanding is that it is fine and we still just get an http context object. No exception is thrown (at least as long as we don't operate on the object). Added an extra guard to check if we are in a http context and otherwise early-return just in case.

Reply:
Adding the check has the non-trivial effect that we will not instrument requests for non-http contexts. I would try and explore other ways of determining whether we already instrumented a request.

Reply:
Changed it